### DISCUSSION TITLE
Functionality of meta-tegra Yocto solution vs. Ubuntu 20.04

### DISCUSSION THREAD

**Discussion by gkenaley** [OE4T/meta-tegra#1117] [General]
A user is interested in developing AI applications on boards such as the Jetson Orin system.
He's wondering which is more suitable for production embedded system use: Yocto/meta-tegra or Ubuntu 20.04.
He had heard that meta-tegra is using older Nvidia packages and isn't as complete nor as Nvidia-endorsed as Ubuntu.
What's the straight story here? Is meta-tegra ready for high-volume deployment?
Thanks!

**Comment by dwalkes** [comment]
>  meta-tegra is using older Nvidia packages

I'm not sure what this is referring to.

> nor as Nvidia-endorsed as Ubuntu.

NVIDIA doesn't directly support oe4t/meta-tegra so we do rely on community support.  There are commercial support options avaialble for meta-tegra.  Speaking from personal experience I've had much more help from the community here than I have from NVIDIA.  Perhaps if you are shipping a million devices per year the experience could be different.

> Is meta-tegra ready for high-volume deployment?

There are definitely folks who are using in production today.

Ultimately you need to evaluate the options and decide what's best for your application based on you and your team's goals/skills/desires/personal bias.

**Comment by madisongh** [comment]
>>meta-tegra is using older Nvidia packages
>
>I'm not sure what this is referring to.

Me either.  We stay up-to-date with NVIDIA's public releases, and don't usually lag behind more than a couple of weeks.  You'll see notes on each of the recent L4T/JetPack releases on [the wiki](https://github.com/OE4T/meta-tegra/wiki).

And as for other, non-NVIDIA open-source packages you might rely on, the most recent LTS branch of OE-Core is certainly newer than anything in the Ubuntu 20.04 base that L4T/JetPack uses.

But as Dan says, it depends on the needs of your product/application, team, etc.  Building your own Linux distro with the Yocto Project tools gives you a lot of control over what goes into your product, but does have a learning curve.
